base/db/R/dbfiles.R

```diff
@@ -661,4 +661,248 @@ dbfile.id <- function(type, file, con, hostname=PEcAn.remote::fqdn()) {
+  if (dim(dbfiles)[1] == 0 & file.pattern == "*.clim" | files.changed > 0 & file.pattern == "*.clim" ){ 
+    
+    #Recheck what files are in the directory since others may have been moved above 
+    old.files <- list.files(path= old.dir, pattern = file.pattern)
+    
+    #Recreate full file path 
+    full.old.file = file.path(old.dir, old.files)
+    
+    #Record number of files that will have a symbolic link made 
+    files.sym = length(full.old.file)
+    
+    #Error check again to make sure there aren't any matching dbfiles 
+    dbfile.path = dirname(full.old.file)
+    dbfiles <- dplyr::tbl(con, "dbfiles") %>% dplyr::collect() %>%
+      dplyr::filter(file_name %in% basename(full.old.file)) %>% 
+      dplyr::filter(file_path %in% dbfile.path)
+    
+    if(dim(dbfiles)[1] > 0){ 
+      PEcAn.logger::logger.error("There are still dbfiles matching these files! Canceling symbolic link creation")
+      error = 1
+    }
+    
+    #Create file path for symbolic link 
+    full.new.file = file.path(new.dir, old.files)
+    
+    #Line up files 
+    full.new.file = sort(full.new.file)
+    full.old.file <- sort(full.old.file)
+    
+    #Check to make sure the files are the same length 
+    if(length(full.new.file) != length(full.old.file)) { 
+      
+      PEcAn.logger::logger.error("Files to be moved don't match up with BETY. Canceling Move")
+      error = 1
+    }
+    
+    #Move file and create symbolic link if there are no errors 
+    
+    if(error ==0){
+      for(i in 1:length(full.old.file)){
+        fs::file_move(full.old.file[i], new.dir)
+        R.utils::createLink(link = full.old.file[i], target = full.new.file[i])
+      }#end i loop 
+    } #end error loop 
+    
+    
+  } #end clim if statement 
+  
+  
+  #If files are .nc files and aren't in BETY for some reason, we will register them 
+  if (dim(dbfiles)[1] == 0 & file.pattern == "*.nc" | files.changed == 1 & file.pattern == "*.nc" ){ 
+    
+    #Re make full file path and find files that were not moved 
+    old.files <- list.files(path= old.dir, pattern = file.pattern)
+    
+    full.old.file = file.path(old.dir, old.files)
+    
+    #Record how many files are being registered to BETY 
+    files.reg= length(full.old.file)
+    
+    #Check again to make sure there aren't any matching dbfiles 
+    dbfile.path = dirname(full.old.file)
+    dbfiles <- dplyr::tbl(con, "dbfiles") %>% dplyr::collect() %>%
+      dplyr::filter(file_name %in% basename(full.old.file)) %>% 
+      dplyr::filter(file_path %in% dbfile.path)
+    
+    if(dim(dbfiles)[1] > 0){ 
+      PEcAn.logger::logger.error("There are still dbfiles matching these files! Canceling symbolic link creation")
+      error = 1
+    }
+    
+    if(error == 0){
+      
+      for(i in 1:length(full.old.file)){
+        
+        file_path = dirname(full.old.file[i])
+        file_name = basename(full.old.file[i])
+        
+        
+        dbfile.input.insert(in.path = file_path,
```

Comment:
Are you asking that I change it or that I document it more? I did it this way because you said that SIPNET .clim files aren't always registered and that I should make a symbolic link instead.

Reply:
Both. Consider "move but not register" to be a general concept that could be handled by a function argument (that defaults to "move and register"), not something tied to the "clim" file format specifically. Likewise, "move and register" isn't tied to netcdf files specifically.